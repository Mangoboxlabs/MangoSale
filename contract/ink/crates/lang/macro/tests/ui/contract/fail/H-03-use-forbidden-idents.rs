use ink_lang as ink;

#[ink::contract]
mod forbidden_indents {
    #[ink(storage)]
    pub struct ForbiddenIndents {}

    impl ForbiddenIndents {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        /// An ink! message starting with __ink_ prefix.
        #[ink(message)]
        pub fn __ink_message(&self) {
            // All identifiers starting with `__ink_` are forbidden to use in ink!.
            let __ink_first = ();
        }
    }
}

fn main() {}
