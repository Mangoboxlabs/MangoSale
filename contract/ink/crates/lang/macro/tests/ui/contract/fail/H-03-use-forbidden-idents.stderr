error: encountered invalid identifier starting with __ink_
  --> $DIR/H-03-use-forbidden-idents.rs:16:16
   |
16 |         pub fn __ink_message(&self) {
   |                ^^^^^^^^^^^^^

error: encountered invalid identifier starting with __ink_
  --> $DIR/H-03-use-forbidden-idents.rs:18:17
   |
18 |             let __ink_first = ();
   |                 ^^^^^^^^^^^
